src/attr.rs

```diff
@@ -0,0 +1,44 @@
+use syn::{self, Meta};
+use quote::ToTokens;
+
+/// Defines `#[attribute]`s.
+#[derive(Debug, Clone)]
+pub enum Attribute {
+    /// Attribute in form of a call: `#[call(args)]`
+    Call {
+        path: String,
+        args: Vec<String>,
+    },
+
+    /// Attributes in other forms which aren't calls
+    Other {
+        tokens: String,
+    },
+}
+
+pub(crate) fn convert_attributes(attrs: &[syn::Attribute]) -> Vec<Attribute> {
```

Comment:
How about keeping them as syn::Attribute? I don't see a reason to do this conversion.

Reply:
It didn't seem to me as straightforward as I wanted it to be to work with the field attributes from the perspective of a macro author so I decided to abstract implementation details (`syn::Attribute`) away with this.

Reply:
I don't have strong opinions either way. I see eupn's point of making it easier for the macro author for the simple case, but syn::Attribute offers a bit more flexibility. I get that Other is a catch all and can be parsed further, but with syn::Attribute, there is already some added structure.

Reply:
Got you guys, I'll change that.